File: ArraysAndHashing/product-of-array-except-self/product-of-array-except-self.cs
public class Solution
{
    public int[] ProductExceptSelf(int[] nums)
    {
        int[] res = new int[nums.Length];
        int prefix = 1;
        int postfix = 1;

        for (int i = 0; i < nums.Length; i++)
        {
            res[i] = prefix;
            prefix *= nums[i];
        }

        for (int i = nums.Length - 1; i >= 0; i--)
        {
            res[i] *= postfix;
            postfix *= nums[i];
        }

        return res;
    }
}

File: ArraysAndHashing/product-of-array-except-self/product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function (nums) {
    const res = new Array(nums.length);
    let prefix = 1;
    let postfix = 1;

    for (let i = 0; i < nums.length; i++) {
        res[i] = prefix;
        prefix *= nums[i];
    }

    for (let i = nums.length - 1; i >= 0; i--) {
        res[i] *= postfix;
        postfix *= nums[i];
    }

    return res;
};
